Modularize ProfileView by Extracting Profile Logic into Composables

mohi-devhub opened the pull request:
### Summary
Refactored the monolithic `ProfileView.vue` into smaller, focused components and composables, improving **maintainability**, **testability**, and **separation of concerns** without changing any UI or behavior.

### Why
`ProfileView.vue` had grown to ~940 lines and mixed UI, state, and Firebase logic, making it:
* Hard to read and maintain
* Difficult to test
* Bug-prone (critical provider detection issue in delete flow)

### What Changed

#### 1. Component Extraction
`ProfileView.vue` now acts as a lightweight orchestrator.

**New components:**
* `UserProfileCard` – profile sidebar display
* `EditProfileDialog` – edit profile modal & image upload
* `ChangePasswordForm` – password update with live validation
* `DeleteAccountSection` – multi-step account deletion flow

Each component has a clear responsibility and a props-based API.

#### 2. Business Logic → Composables
Extracted logic into reusable composables:
* `useProfile` – profile fetch/update, image upload, errors
* `useChangePassword` – password rules & validation state
* `useDeleteAccount` – re-authentication & deletion flow

#### 3. Bug Fix
Fixed incorrect provider detection in delete account flow:

```javascript
// before (incorrect)
user.providerData.includes(a => a.providerId !== 'google.com')

// after (correct)
user.providerData.some(p => p.providerId === 'google.com')
```

### Impact

| Metric | Before | After |
| :--- | :--- | :--- |
| **ProfileView.vue** | 941 lines | **156 lines** |
| Components | 1 | **5** |
| Composables | 0 | **3** |

✅ No breaking changes

✅ No visual/UI changes

✅ All functionality preserved

---

### Verification

Manually tested:
* Profile load & edit
* Avatar upload
* Password change validation
* Account deletion (Google + Email/Password)
* No console or lint errors

---

### Files Structure
src/features/auth/
├── components/
│   ├── ChangePasswordForm.vue        [NEW]
│   ├── DeleteAccountSection.vue      [NEW]
│   ├── EditProfileDialog.vue         [NEW]
│   ├── UserProfileCard.vue           [NEW]
│   └── GoogleSignInButton.vue        [existing]
├── composables/
│   ├── useChangePassword.js          [NEW]
│   ├── useDeleteAccount.js           [NEW]
│   └── useProfile.js                 [NEW]
└── views/
    └── ProfileView.vue               [REFACTORED: 941→156 lines]

**Benefits**
* Clear separation of concerns
* Easier testing and reuse
* Smaller, readable components
* Fixed critical auth provider bug
* Improved developer experience

Fixes : #1064

mohi-devhub committed:
refactor: extract profile management composables

- Created useProfile.js for profile CRUD operations
- Created useChangePassword.js for password change logic
- Created useDeleteAccount.js for account deletion flow
- Encapsulated Firebase auth and Firestore logic
- Added proper error handling and toast notifications

mohi-devhub committed:
refactor: extract UserProfileCard component

- Created self-contained UserProfileCard component
- Displays user avatar, name, and profile details
- Accepts userprofile, userEmail, and loading props
- Maintains consistent UI with original design

mohi-devhub committed:
chore: analyze ProfileView.vue responsibilities

- Backup original 941-line ProfileView.vue
- Identified responsibilities: profile display, editing, password change, account deletion
- Planned component extraction strategy

mohi-devhub committed:
refactor: extract useProfile composable

- Extract profile fetching, updating, and image upload logic
- Centralize profile state management
- Remove profile logic duplication from view

mohi-devhub committed:
refactor: extract useChangePassword composable

- Extract password change logic with validation rules
- Implement live password strength indicators
- Encapsulate password form state and handlers

mohi-devhub committed:
refactor: extract useDeleteAccount composable

- Extract 2-step account deletion flow with re-authentication
- Handle both Google and email/password providers
- Centralize deletion state management and error handling

mohi-devhub committed:
refactor: extract EditProfileDialog component

- Create modal dialog for profile editing
- Handle profile image upload with preview
- Include form validation for username, contact, and country
- Emit events for save and cancel actions

mohi-devhub committed:
refactor: extract ChangePasswordForm component

- Create self-contained password change form
- Integrate useChangePassword composable
- Display live password validation feedback
- Handle form submission and reset

mohi-devhub committed:
refactor: extract DeleteAccountSection component

- Create delete account section with 2-step confirmation
- Integrate useDeleteAccount composable
- Handle both Google and email/password re-authentication
- Manage dialog state and reset on close

mohi-devhub committed:
refactor: clean up ProfileView.vue orchestrator

- Reduce from 941 lines to 157 lines (83% reduction)
- Import and use extracted components
- Delegate all business logic to composables
- Maintain layout and tab structure
- Keep only high-level orchestration logic

mohi-devhub committed:
chore: remove unused component files

- Remove ProfileAccountTab.vue (replaced by inline form + EditProfileDialog)
- Remove ProfileSecurityTab.vue (replaced by ChangePasswordForm + DeleteAccountSection)
- Remove ProfileDeleteAccountDialog.vue (replaced by DeleteAccountSection)

mohi-devhub committed:
chore: remove ProfileView.vue.backup file

- Backup file not needed in repository
- Original file preserved in git history

mohi-devhub committed:
fix: resolve SonarCloud warnings - use globalThis and remove unused import

mohi-devhub committed:
style: fix ESLint formatting issues

mohi-devhub committed:
fix: resolve ESLint errors

jvJUCA reviewed (changes requested):
- Remove linting files changes
- check the error in the security tab
<img width="1843" height="913" alt="image" src="https://github.com/user-attachments/assets/7deebb4b-bcc3-4587-8a22-aeae743aa4e3" />

mohi-devhub committed:
fix: resolve validation rules and i18n issues in password form

mohi-devhub committed:
Revert "fix: resolve ESLint errors"

This reverts commit 65b6049419a73469a3e87ffd57830e31e0991c29.

mohi-devhub committed:
Revert "style: fix ESLint formatting issues"

This reverts commit fa12ac91d55340c4fa67b69de66da2b05951e633.

mohi-devhub committed:
Fix Google OAuth password change and profile image upload

- Add Google OAuth detection for password changes
- Skip current password requirement for Google users
- Add re-authentication via Google popup for OAuth users
- Fix profile image not persisting after upload
- Add image compression (resize to 800px, 80% quality)
- Add loading states and better error handling
- Add profile refetch on dialog close
- Add fallback icons for empty profile images
- Update translations for Google user password flow

jvJUCA reviewed (approved):
Congratulations! 🎉 Your pull request has been accepted! Thank you for your valuable contribution.

Your changes have been merged successfully and will be included in the develop branch. Your effort in improving our project is greatly appreciated.

Thanks again for your contribution, and we look forward to your continued involvement in our project!

Best regards,
@jvJUCA

jvJUCA committed:
Merge branch 'develop' into refactor/profile-view-modular-components
